pkg/gohai/processes/processes.go

```diff
@@ -3,27 +3,65 @@
 // Copyright © 2015 Kentaro Kuribayashi <kentarok@gmail.com>
 // Copyright 2014-present Datadog, Inc.
 
-//go:build linux || darwin
-// +build linux darwin
-
-// Package processes regroups collecting information about existing processes
+// Package processes regroups collecting information about running processes.
 package processes
 
 import (
+	"flag"
 	"strings"
 	"time"
-
-	"github.com/DataDog/datadog-agent/pkg/gohai/processes/gops"
 )
 
+var options struct {
+	limit int
+}
+
+// Processes is the Collector type of the processes package.
+type Processes struct{}
+
+const name = "processes"
+
+func init() {
+	flag.IntVar(&options.limit, name+"-limit", 20, "Number of process groups to return")
+}
+
+// Name returns the name of the package
+func (processes *Processes) Name() string {
+	return name
+}
+
+// ProcessGroup represents the information about a single process group
+type ProcessGroup struct {
+	// Usernames is the sorted list of usernames of running processes in that groups.
+	Usernames []string
+	// PctCPU is the percentage of cpu used by the group.
+	PctCPU int
+	// PctMem is the percentage of memory used by the group.
+	PctMem float64
+	// VMS is the vms of the group.
+	VMS uint64
+	// RSS is the RSS used by the group.
+	RSS uint64
+	// Name is the name of the group.
+	Name string
+	// Pids is the list of pids in the group.
+	Pids []int32
+}
+
+// Get returns a list of process groups information or an error
+func Get() ([]ProcessGroup, error) {
+	return getProcessGroups(options.limit)
+}
+
 // ProcessField is an untyped representation of a process group,
 // compatible with the legacy "processes" resource check.
 type ProcessField [7]interface{}
 
-// getProcesses return a JSON payload which is compatible with
-// the legacy "processes" resource check
-func getProcesses(limit int) ([]interface{}, error) {
-	processGroups, err := gops.TopRSSProcessGroups(limit)
+// Collect collects the processes information.
+// Returns an object which can be converted to a JSON or an error if nothing could be collected.
+// Tries to collect as much information as possible.
+func (processes *Processes) Collect() (result interface{}, err error) {
```

Comment:
we don't use the returned variable names anywhere.

```suggestion
func (processes *Processes) Collect() (interface{}, error) {
```

Reply:
Nice catch !